script.js

```diff
@@ -87,29 +88,47 @@ function drawSnake(snakeBlocks) {
 
 function die() {
   dead = true;
-  let button = createButton("Restart");
   console.log("no");
   let dieTxt = 'Bruh';
+  dieButton();
   fill(220, 20, 60);
   textSize(30);
   text(dieTxt, 255, 280);
+  frameRate(0);
+}
+
+function dieButton(){
+  let button = createButton("Restart");
   button.id = 'restart';
   button.size(200, 100);
   button.position(200, 320);
   button.style("font-family", "Bodoni");
   button.style("font-size", "48px");
   button.mouseClicked(restartGame);
-  frameRate(0);
 }
 
 function restartGame() {
-  var b = document.getElementsByTagName("button");
+  let b = document.getElementsByTagName("button");
   b[jankyCounter].style.display = "none";
   setup();
   jankyCounter++;
 }
 
 function touchingSelf() {
+ let newPose = newPos(direction, snakeBlocks);
+  if (checkBody(snakeBlocks, newPose)) {
+    die();
+  }
+}
+
+function touchingEdge(direction, snakeBlocks) {
+  let newPose = newPos(direction, snakeBlocks);
+  if (newPose[0] >= 30 || newPose[0] < 0 || newPose[1] >= 30 || newPose[1] < 0) {
+    die();
+  }
+}
+
+function newPos(direction,snakeBlocks){
```

Comment:
This function contains two bits of code that are repeated a lot: `snakeBlock[snakeBlocks.length - 1][0]` and ``snakeBlock[snakeBlocks.length - 1][0]`. And within them obviously there's the repeated `snakeBlock[snakeBlocks.length - 1]`. Can you extract those values at the beginning of this function and put them in variables that make the code more readable?